src/tools/r2rdump/R2RMethod.cs

```diff
@@ -204,27 +207,24 @@ public R2RMethod(byte[] image, MetadataReader mdReader, uint rid, int entryPoint
             SignatureHeader signatureHeader = signatureReader.ReadSignatureHeader();
             IsGeneric = signatureHeader.IsGeneric;
             GenericParameterHandleCollection genericParams = _methodDef.GetGenericParameters();
-            _genericParamInstanceMap = new Dictionary<string, GenericInstance>();
+            _genericParamInstanceMap = new Dictionary<string, string>();
             
             int argCount = signatureReader.ReadCompressedInteger();
             if (IsGeneric)
             {
                 argCount = signatureReader.ReadCompressedInteger();
             }
 
-            ReturnType = new SignatureType(ref signatureReader, mdReader, genericParams);
-            ArgTypes = new SignatureType[argCount];
-            for (int i = 0; i < argCount; i++)
-            {
-                ArgTypes[i] = new SignatureType(ref signatureReader, mdReader, genericParams);
-            }
-
+            DisassemblingTypeProvider provider = new DisassemblingTypeProvider();
             if (IsGeneric && instanceArgs != null && tok != null)
             {
                 InitGenericInstances(genericParams, instanceArgs, tok);
             }
+            
+            DisassemblingGenericContext genericContext = new DisassemblingGenericContext(new string[0], _genericParamInstanceMap.Values.ToArray());
```

Comment:
It's hard to visualize how this works at the moment, but it looks good that you're able to use the DisassemblingGenericContext. If the results are satisfactory then 👍